backend/molgenis-emx2-tasks/src/main/java/org/molgenis/emx2/tasks/TaskServiceInMemory.java

```diff
@@ -44,12 +40,17 @@ public Task getTask(String id) {
     return tasks.get(id);
   }
 
+  @Override
+  public Collection<Task> listTasks() {
+    return tasks.values();
+  }
+
   @Override
   public void removeOlderThan(long milliseconds) {
     Set<String> keys = tasks.keySet();
```

Comment:
forEach ?

Reply:
?

Reply:
why get a key list and then do lots of lookups , instead of applying function to each item , the whole for the the task.get(index).prop makes it hard to read, could be tasks.forEach(doSomething), or in case of remove https://docs.oracle.com/javase/8/docs/api/java/util/Collection.html#removeIf-java.util.function.Predicate-